cpp/src/io/avro/reader_impl.cu

```diff
@@ -270,100 +269,76 @@ rmm::device_buffer decompress_data(datasource& source,
     // comp_block_data contains contents of the avro file starting from the first block, excluding
     // file header. meta.block_list[i].offset refers to offset of block i in the file, including
     // file header.
-    // Find ptrs to each compressed block in comp_block_data by removing header offset.
-    cudf::detail::hostdevice_vector<void const*> compressed_data_ptrs(num_blocks, stream);
+    cudf::detail::hostdevice_vector<device_span<uint8_t const>> compressed_blocks(num_blocks,
+                                                                                  stream);
     std::transform(meta.block_list.begin(),
                    meta.block_list.end(),
-                   compressed_data_ptrs.host_ptr(),
+                   compressed_blocks.host_ptr(),
                    [&](auto const& block) {
-                     return static_cast<std::byte const*>(comp_block_data.data()) +
-                            (block.offset - meta.block_list[0].offset);
+                     // Find ptrs to each compressed block by removing the header offset
+                     return device_span<uint8_t const>{
```

Comment:
Looks like this went from `std::byte` to `uint8_t`. That's generally the opposite of the direction we want to go, right? Can we use `std::byte`?

Reply:
The decompress API takes spans of `uint8_t`, so that's what I used.
Previous cast was done only to enable pointer arithmetic, as we were passing an array of void pointers to nvcomp.